src/compiler/checker.ts

```diff
@@ -9801,8 +9888,13 @@ namespace ts {
             //
             // For a decorator, no arguments are susceptible to contextual typing due to the fact
             // decorators are applied to a declaration by the emitter, and not to an expression.
+            let excludeCallee: boolean;
             let excludeArgument: boolean[];
             if (!isDecorator) {
+                const calleeNode = node.kind === SyntaxKind.CallExpression && (<PropertyAccessExpression>(<CallExpression>node).expression).expression;
+                if (calleeNode && isContextSensitive(calleeNode)) {
+                    excludeCallee = true;
+                }

```

Comment:
Nope. It fails. I'll take all the contextual typing code for this arguments out.